### PR TITLE
[daikin][wip] add energy readings

### PR DESCRIPTION
**Pull request by gitwouter**
Adding support for energy usage reading to the binding

Reading the aircon/get_week_power_ex endpoint to get energy usage reading for heating & cooling for today, this week & last week.

### CODE REVIEW

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfo.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        // week_heat=<today>/<today-1>/<today-2>/<today-3>/...
+        Map<String, String> responseMap = Arrays.asList(response.split(",")).stream().filter(kv -> kv.contains("="))
+                .map(kv -> {
+                    String[] keyValue = kv.split("=");
+                    String key = keyValue[0];
+                    String value = keyValue.length > 1 ? keyValue[1] : "";
+                    return new String[] { key, value };
+                }).collect(Collectors.toMap(x -> x[0], x -> x[1]));
+
+        if (responseMap.get("ret") != null && responseMap.get("ret").contentEquals("OK")) {
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfo.java:64]: Swap parameters and null check can be omitted:
```suggestion
        if ("OK".equals(responseMap.get("ret"))) {
```

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfo.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+            // get the heating info
+            String[] heatingValues = responseMap.get("week_heat").split("/");
+            info.energyHeatingToday = Optional.of(Double.parseDouble(heatingValues[0]) / 10);
+            double thisWeekEnergy = 0;
+            for (int i = 0; i < thisWeekLastDayIndex; i += 1) {
+                thisWeekEnergy += Integer.parseInt(heatingValues[i]);
+            }
+            double previousWeekEnergy = 0;
+            for (int i = thisWeekLastDayIndex; i < thisWeekLastDayIndex + 7; i += 1) {
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfo.java:76]:
```suggestion
            for (int i = thisWeekLastDayIndex; i < thisWeekLastDayIndex + 7; i++) {
```
and same comment for other loops.

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinBaseHandler.java [hunk 3]
```diff
@@ -184,6 +186,12 @@ protected void updateTemperatureChannel(String channel, Optional<Double> maybeTe
                 .orElse(UnDefType.UNDEF));
     }
 
+    protected void updateEnergyChannel(String channel, Optional<Double> maybeEnergy) {
+        updateState(channel,
+                maybeEnergy.<State>map(t -> new QuantityType<>(new DecimalType(t), SmartHomeUnits.KILOWATT_HOUR))
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/handler/DaikinBaseHandler.java:191]:
```suggestion
                maybeEnergy.<State>map(t -> new QuantityType<>(t, SmartHomeUnits.KILOWATT_HOUR))
```

File: bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfo.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    public static EnergyInfo parse(String response) {
+
+        EnergyInfo info = new EnergyInfo();
+
+        logger.debug("parsing energy {}", response);
+
+        // /aircon/get_week_power_ex
+        // ret=OK,s_dayw=0,week_heat=1/1/1/1/1/5/2/1/1/1/1/2/1/1,week_cool=0/0/0/0/0/0/0/0/0/0/0/0/0/0
+        // week_heat=<today>/<today-1>/<today-2>/<today-3>/...
+        Map<String, String> responseMap = Arrays.asList(response.split(",")).stream().filter(kv -> kv.contains("="))
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfo.java:56]: You should use `InfoParser.parse(response);` The `InfoParser` class also has some new methods: `parseArrayofInt` that you could use to parse the specific values below. That might simplify the code too.

Comment (gitwouter) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfo.java:56]: I should remove this pull request. While I was in discussion/feedback on this PR another one was merged and made my work obsolete. Cf. https://community.openhab.org/t/daikin-binding-logging-energy-usage/96964

Comment (Hilbrand) [reply bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/api/EnergyInfo.java:56]: I might be wrong, but if you're referring to #7708. It doesn't look like that pr overlapped with your pr. That pr added monthly reports, while you added daily/weekly data. That seems other data. So I think your changes are still relevant and from a quick look that pr added some methods that might could make your code simpler by using the added helper methods for parsing the data.

**Pull request closed without merging**